feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/widget/SessionsItemDecoration.kt

```diff
@@ -45,37 +48,34 @@ class SessionsItemDecoration(
             val view = parent.getChildAt(i)
             val position = parent.getChildAdapterPosition(view)
             if (position == -1 || position >= groupAdapter.itemCount) return
-            val item = groupAdapter.getItem(position)
-            val previousItem = if (position - 1 < 0) {
-                null
-            } else {
-                groupAdapter.getItem(position - 1)
-            }
-            if (item is SessionItem) {
-                val time = item.session.startTime.toString("HH:mm")
-                val previousTime = if (previousItem is SessionItem) {
-                    previousItem.session.startTime.toString("HH:mm")
-                } else {
-                    null
-                }
 
-                if (lastTime == time) continue
+            val time = getSessionTime(position) ?: continue
 
-                lastTime = time
+            if (lastTime == time) continue
+            lastTime = time
 
-                val yPosition = if (view.top < 0 || time == previousTime) {
-                    textSize
-                } else {
-                    view.top + textSize
-                }
+            val nextTime = getSessionTime(position + 1)
 
-                c.drawText(
-                    time,
-                    textLeftSpace.toFloat(),
-                    yPosition.toFloat(),
-                    paint
-                )
+            var textY = view.top.coerceAtLeast(0) + textPaddingTop + textSize
+            if (time != nextTime) {
+                textY = textY.coerceAtMost(view.bottom - textPaddingBottom)
             }
+
+            c.drawText(
+                time,
+                textLeftSpace.toFloat(),
+                textY.toFloat(),
+                paint
+            )
         }
     }
+
+    private fun getSessionTime(position: Int): String? {
+        if (position < 0 || position >= groupAdapter.itemCount) {
+            return null
+        }
+
+        val item = groupAdapter.getItem(position) as? SessionItem
```

Comment:
How about using elvis operator?♻️

groupAdapter.getItem(position) as? SessionItem ?: return null

Reply:
Sure! Actually, I was torn between current code and your proposal then selected current one... Will fix.

Reply:
Thanks! I will sleep 💤 Good night!

Reply:
Thank you for the swift review even midnight (or early morning)... Good night!

Reply:
Fixed at 4f54446d112ea4cdd233e9f9321583c9dad50e58.